src/main/java/uk/gov/ida/eventemitter/EventEmitter.java

```diff
@@ -23,8 +26,9 @@ public void record(final Event event) {
         if (event != null) {
             String encryptedEvent = null;
             try {
-                encryptedEvent = encrypter.encrypt(event);
-                eventSender.sendAuthenticated(event, encryptedEvent);
+                final Event hashedEvent = eventHasher.hash(event);
```

Comment:
I think the object creation is quite expensive here and we should just re-assign to the "pid" field - I just didn't want it to happen as a side effect in the hash method. Maybe this method should be `eventHasher.getHashedPid(String idpEntityId, String pid)` instead?

Reply:
Re-assigning a hashed persistent id to the "pid" field is still a mutable operation (i.e. it still modifies the Event object to have the hashed persistent id). The eventHasher.getHashedPid(String idpEntityId, String pid) method simply returns the hashed persistent id and it does not make the Event object to have the hashed persistent id. Do you want me to revert the hashing method back to the mutable method again to avoid expensive object creation operations? I mentioned earlier that the mutable method avoids expensive object creation operations.